llvm/test/tools/sycl-post-link/spec-constants/nested-struct.ll

```diff
@@ -0,0 +1,149 @@
+; This test demonstrates that multiple padding elements can be
```

Comment:
Yea, with the way the metadata is handled right now, they are emitted for nested structs. I added this because a previous assertion `assert(Result.size() <= IDs.size() + 1);` that you added in a WIP commit was not necessarily satisfied.